### PR TITLE
initial lint, unit test and codeql github actions

### PR DESCRIPTION
**Pull request by hanstrompert**
- initial set of GitHub workflows
  - isort, black, flake8 and mypy
  - pytest with codecov
  - CodeQL analysis
- disable mypy step for now until typing is all fixed
- pre-commit exclude uses regex not fileglob
- fix flake8 errors
- black now ignores generated files in grpc_nsi and build
- isort now ignores generated files in grpc_nsi
- run lint tools and pytest using add dependencies

### CODE REVIEW

File: .github/workflows/run-linting-tests.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      - name: Lint with flake8
+        run: |
+          # stop the build if there are Python syntax errors or undefined names
+          flake8 .
+      #
+      # disabled mypy for now, FIXME
+      #
+      # - name: Check with mypy
+      #  run: |
+      #    mypy src/supa
```
Comment (pboers1988) [inline .github/workflows/run-linting-tests.yml:42]: What are the errors? It should be relatively simple to fix them, with the correct type: ignore statements, or they really do indicate a bug.

**Review by pboers1988 — CHANGES_REQUESTED** [review]
It should be relatively easy to either type: ignore the mypy errors or fix them, they usually really indicate a bug, and turning type checking off would not be preferred.

File: .github/workflows/run-linting-tests.yml (hunk 1 above)
Comment (pboers1988) [reply .github/workflows/run-linting-tests.yml:42]: Judging by the last failed run, you should be able to simply exclude main in the run, and the other errors seem to indicate valid potential bugs.

**Review by acidjunk — APPROVED** [review]
Nice github actions. There is on other action that you can copy from the core: it will open a PR automatically upon the first push to a new branch.

Not a show stopper but nice to have.

File: .github/workflows/run-linting-tests.yml (hunk 1 above)
Comment (hanstrompert) [reply .github/workflows/run-linting-tests.yml]: Good point. Fixing the typing errors wasn't really hard, even without ignoring stuff. Enabled mypy again.

**Review by pboers1988 — APPROVED** [review]

**Pull request merged**